components/browser/engine-system/src/main/java/mozilla/components/browser/engine/system/NestedWebView.kt

```diff
@@ -46,18 +48,15 @@ class NestedWebView(context: Context) : WebView(context), NestedScrollingChild {
     internal var childHelper: NestedScrollingChildHelper = NestedScrollingChildHelper(this)
 
     /**
-     * Integer indicating how user's MotionEvent was handled.
+     * How user's MotionEvent will be handled.
      *
-     * There must be a 1-1 relation between this values and [EngineView.InputResult]'s.
-     */
-    @Suppress("Deprecation")
-    internal val inputResult: Int
-        get() = getInputResult(eventHandled)
-
-    /**
-     * Holder for if the previous [android.view.MotionEvent] was handled by us or not.
+     * @see InputResultDetail
      */
-    private var eventHandled: Boolean = false
+    @VisibleForTesting
+    internal val inputResultDetail = InputResultDetail().apply {
+        // The constructor enables vertical overscrolling by default. Don't want that.
+        update(overscrollDirections = 0)
```

Comment:
nit: Should we use a constant instead of a direct value?

Reply:
This is now just
>     internal var inputResultDetail = InputResultDetail.newInstance()